Intermittent test failure in /lib/es/tests/test_commands.py:test_reindexation_starting_from_one_addon()

I have seen this at least a couple of times on recent PR Travis runs:

```
=================================== FAILURES ===================================

__________ TestIndexCommand.test_reindexation_starting_from_one_addon __________

src/olympia/lib/es/tests/test_commands.py:165: in test_reindexation_starting_from_one_addon

    self._test_reindexation()

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

src/olympia/lib/es/tests/test_commands.py:133: in _test_reindexation

    self.check_results(self.expected)

        ReindexThread = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand._test_reindexation.<locals>.ReindexThread'>

        old_addons_count = 1

        old_indices = [('test_amo_addons-20200602162433', 'test_amo_addons'),

 ('test_amo_addons_stats_download_counts-20200602162433',

  'test_amo_addons_stats_download_counts'),

 ('test_amo_addons_stats_update_counts-20200602162433',

  'test_amo_addons_stats_update_counts')]

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

        t          = <ReindexThread(Thread-1, stopped 140473940182784)>

        wipe       = False

src/olympia/lib/es/tests/test_commands.py:79: in check_results

    assert addon.pk in got, '%s is not in %s' % (addon.pk, got)

E   AssertionError: 1 is not in []

E   assert 1 in []

E    +  where 1 = <Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>.pk

        addon      = <Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>

        expected   = [<Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>,

 <Addon: 2: Addôn 1aa2878bb3604c31b44f4aba997316cc>]

        got        = []

        response   = <Response status_code=200, "application/json">

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

------------------------------ Captured log call -------------------------------

INFO     z.elasticsearch:reindex.py:79 Returning reindexing group for test_amo_addons-20200602162717

INFO     z.elasticsearch:reindex.py:57 Create the index test_amo_addons-20200602162717, for alias: test_amo_addons

INFO     z.elasticsearch:reindex.py:64 Flagging the database to start the reindexation

INFO     z.elasticsearch:reindex.py:50 Rebuilding aliases with actions: [{'remove': {'index': 'test_amo_addons-20200602162433', 'alias': 'test_amo_addons'}}, {'add': {'index': 'test_amo_addons-20200602162717', 'alias': 'test_amo_addons'}}]

INFO     z.elasticsearch:reindex.py:71 Unflagging the database

INFO     z.elasticsearch:reindex.py:44 Removing indices 'test_amo_addons-20200602162433'
```

Traceback for theo ther ES test fail:
```
_________________ TestIndexCommand.test_reindexation_with_wipe _________________

src/olympia/lib/es/tests/test_commands.py:172: in test_reindexation_with_wipe

    self._test_reindexation(wipe=True)

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_with_wipe>

src/olympia/lib/es/tests/test_commands.py:132: in _test_reindexation

    self.refresh()

        ReindexThread = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand._test_reindexation.<locals>.ReindexThread'>

        old_addons_count = 1

        old_indices = [('test_amo_addons_stats_download_counts-20200609080209',

  'test_amo_addons_stats_download_counts'),

 ('test_amo_addons_stats_update_counts-20200609080209',

  'test_amo_addons_stats_update_counts')]

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_with_wipe>

        t          = <ReindexThread(Thread-3, stopped 139937182476032)>

        wipe       = True

src/olympia/amo/tests/__init__.py:925: in refresh

    cls.es.indices.refresh(settings.ES_INDEXES[index])

        cls        = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand'>

        index      = 'default'

.tox/es/lib/python3.6/site-packages/elasticsearch/client/utils.py:73: in _wrapped

    return func(*args, params=params, **kwargs)

        args       = (<elasticsearch.client.indices.IndicesClient object at 0x7f45ae18a0b8>,

 'test_amo_addons')

        es_query_params = ('allow_no_indices',

 'expand_wildcards',

 'force',

 'ignore_unavailable',

 'operation_threading')

        func       = <function IndicesClient.refresh at 0x7f45b724e840>

        kwargs     = {}

        p          = 'request_timeout'

        params     = {}

.tox/es/lib/python3.6/site-packages/elasticsearch/client/indices.py:56: in refresh

    '_refresh'), params=params)

        index      = 'test_amo_addons'

        params     = {}

        self       = <elasticsearch.client.indices.IndicesClient object at 0x7f45ae18a0b8>

.tox/es/lib/python3.6/site-packages/elasticsearch/transport.py:312: in perform_request

    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)

        attempt    = 0

        body       = None

        connection = <Urllib3HttpConnection: http://127.0.0.1:9200>

        ignore     = ()

        method     = 'POST'

        params     = {}

        retry      = False

        self       = <elasticsearch.transport.Transport object at 0x7f45ae18a320>

        timeout    = None

        url        = '/test_amo_addons/_refresh'

.tox/es/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:129: in perform_request

    self._raise_error(response.status, raw_data)

        body       = None

        duration   = 0.0020112991333007812

        full_url   = 'http://127.0.0.1:9200/test_amo_addons/_refresh'

        ignore     = ()

        kw         = {}

        method     = 'POST'

        params     = {}

        raw_data   = ('{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"}],"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"},"status":404}')

        response   = <urllib3.response.HTTPResponse object at 0x7f45aae9e358>

        self       = <Urllib3HttpConnection: http://127.0.0.1:9200>

        start      = 1591689896.9294157

        timeout    = None

        url        = '/test_amo_addons/_refresh'

.tox/es/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: in _raise_error

    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)

E   elasticsearch.exceptions.NotFoundError: TransportError(404, 'index_not_found_exception', 'no such index')

        additional_info = {'error': {'index': 'test_amo_addons',

           'index_uuid': '_na_',

           'reason': 'no such index',

           'resource.id': 'test_amo_addons',

           'resource.type': 'index_or_alias',

           'root_cause': [{'index': 'test_amo_addons',

                           'index_uuid': '_na_',

                           'reason': 'no such index',

                           'resource.id': 'test_amo_addons',

                           'resource.type': 'index_or_alias',

                           'type': 'index_not_found_exception'}],

           'type': 'index_not_found_exception'},

 'status': 404}

        error_message = 'index_not_found_exception'

        raw_data   = ('{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"}],"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"},"status":404}')

        self       = <Urllib3HttpConnection: http://127.0.0.1:9200>

        status_code = 404
```

--- a/src/olympia/amo/tests/__init__.py
+++ b/src/olympia/amo/tests/__init__.py
@@ -942,11 +942,7 @@ class ESTestCase(TestCase):
 
     @classmethod
     def refresh(cls, index='default'):
-        if index in settings.ES_INDEXES:
-            target = settings.ES_INDEXES[index]
-        else:
-            target = index
-        cls.es.indices.refresh(target)
+        cls.es.indices.refresh(settings.ES_INDEXES.get(index, index))
 
     @classmethod
     def reindex(cls, model, index='default'):

--- a/src/olympia/amo/tests/__init__.py
+++ b/src/olympia/amo/tests/__init__.py
@@ -942,7 +942,11 @@ class ESTestCase(TestCase):
 
     @classmethod
     def refresh(cls, index='default'):
-        cls.es.indices.refresh(settings.ES_INDEXES[index])
+        if index in settings.ES_INDEXES:
+            target = settings.ES_INDEXES[index]
+        else:
+            target = index
+        cls.es.indices.refresh(target)
 
     @classmethod
     def reindex(cls, model, index='default'):

--- a/src/olympia/lib/es/tests/test_commands.py
+++ b/src/olympia/lib/es/tests/test_commands.py
@@ -130,7 +130,9 @@ class TestIndexCommand(ESTestCase):
                 self.expected.append(addon_factory())
                 connection._commit()
                 connection.clean_savepoints()
-                self.refresh()
+                # We don't know where the search will happen, the reindexing
+                # could be over by now. So force a refresh on *all* indices.
+                self.refresh(None)
                 self.check_results(self.expected)
 
             if len(self.expected) == old_addons_count:

